src/kem.rs

```diff
@@ -167,10 +166,10 @@ pub enum Ct {
 pub enum Ss {
     X25519(x25519::PublicKey),
     P256(p256::PublicKey),
-    Kyber512(Kyber512SharedSecret),
-    Kyber768(Kyber768SharedSecret),
-    Kyber768X25519(Kyber768SharedSecret, x25519::PublicKey),
-    Kyber1024(Kyber1024SharedSecret),
+    Kyber512(KyberSharedSecret),
```

Comment:
We don't need the different variants here anymore. Let's collaps them into a `Kyber` variant. Or do we get something from the typesafety here?

Reply:
I think having type safety here still makes sense; it'd be nice to enforce the fact that `Kyber768X25519` has a `Kyber768SharedSecret` at the type level. I'm mulling over how best to do this ...